src/app.js

```diff
@@ -15,26 +17,38 @@ import updateGrantsRecipients from './lib/updateGrantsRecipients';
 import { logger, auditLogger, requestLogger } from './logger';
 
 const app = express();
-
 const oauth2CallbackPath = '/oauth2-client/login/oauth2/code/';
+const index = fs.readFileSync(path.join(__dirname, 'client', 'index.html')).toString();
+
+const serveIndex = (req, res) => {
+  const noncedIndex = index.replaceAll('__NONCE__', res.locals.nonce);
+  res.set('Content-Type', 'text/html');
+  res.send(noncedIndex);
+};
 
 app.use(requestLogger);
 app.use(express.json({ limit: '2MB' }));
 app.use(express.urlencoded({ extended: true }));
-app.use(helmet({
-  contentSecurityPolicy: {
+
+app.use((req, res, next) => {
+  res.locals.nonce = crypto.randomBytes(16).toString('hex');
+  const cspMiddleware = helmet.contentSecurityPolicy({
     directives: {
       ...omit(helmet.contentSecurityPolicy.getDefaultDirectives(), 'upgrade-insecure-requests', 'block-all-mixed-content', 'script-src', 'img-src', 'default-src'),
       'form-action': ["'self'"],
-      scriptSrc: ["'self'", 'https://touchpoints.app.cloud.gov/touchpoints/7d519b5e.js'],
-      imgSrc: ["'self'", 'data:', 'https://touchpoints.app.cloud.gov'],
-      defaultSrc: ["'self'", 'https://touchpoints.app.cloud.gov/touchpoints/7d519b5e/submissions.json'],
+      scriptSrc: ["'self'", 'https://touchpoints.app.cloud.gov*', '*.googletagmanager.com'],
```

Comment:
I think this should be either https://touchpoints.app.cloud.gov/* or you can possibly just remove it (the survey is just a link now, not a modal)